photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/GameTrainingDriver.scala

```diff
@@ -637,21 +635,34 @@ object GameTrainingDriver extends GameDriver {
         // TODO: Match on this to make it clearer
         val evaluator = models.head._2.get.head._1
         val baseConfig = models.head._3
-        val evaluationFunction = new GameEstimatorEvaluationFunction(estimator, baseConfig, trainingData, testData)
 
         val range = getOrDefault(hyperParameterTuningRange)
-        val ranges = List.fill(evaluationFunction.numParams)(DoubleRange(log(range.start), log(range.end)))
+        val rangeWrapped = DoubleRange(log(range.start), log(range.end))
+
+        val evaluationFunction = new GameEstimatorEvaluationFunction(estimator, baseConfig, trainingData, testData, range)
+        val dimension = evaluationFunction.numParams
 
         val searcher = getOrDefault(hyperParameterTuning) match {
           case HyperparameterTuningMode.BAYESIAN =>
-            new GaussianProcessSearch[GameEstimator.GameResult](ranges, evaluationFunction, evaluator)
+            new GaussianProcessSearch[GameEstimator.GameResult](dimension, evaluationFunction, evaluator)
 
           case HyperparameterTuningMode.RANDOM =>
-            new RandomSearch[GameEstimator.GameResult](ranges, evaluationFunction)
+            new RandomSearch[GameEstimator.GameResult](dimension, evaluationFunction)
         }
         val observations = searcher.convertObservations(models)
 
-        searcher.findWithObservations(getOrDefault(hyperParameterTuningIter), observations)
+        // Wrap priors by a log transformation
+        observations.foreach { case (prior, _) => log.inPlace(prior) }
+
+        // Rescale priors linearly by the range
+        val slope = 1.0 / rangeWrapped.end - rangeWrapped.start
+        val intercept = - range.start / (range.end - range.start)
```

Comment:
May be im getting this wrong. Isnt range = [0.001, 1000]? If so, then the intercept becomes -0.0001/999.999. 
So, the intercept should also be -rangeWrapped.start*slope instead of using range?

Reply:
You are right. It should be
 val intercept = - rangeWrapped.start / (rangeWrapped.end - rangeWrapped.start)